src/treq/response.py

```diff
@@ -91,3 +93,24 @@ def cookies(self):
                 jar.set_cookie(cookie)
 
         return jar
+
+    def check_status(self):
```

Comment:
I feel that this would be better implemented as a free function in the `treq.content` module (and this method just call that function). Right now the functions in `treq.content` work on any `IResponse` provider, while the present `treq.check_status()` function (really an unbound method) will throw `TypeError` if passed an arbitrary `IResponse` (it requires a `treq.response._Response` or subclass thereof). I think that we should preserve that quality with this new interface so that it is still useful when using the raw Agent API.